server/memphis_handlers_schemas.go

```diff
@@ -295,3 +337,148 @@ func (sh SchemasHandler) RemoveSchema(c *gin.Context) {
+	if err != nil {
+		serv.Errorf("CreateNewVersion error: " + err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+
+	versionNumber := countVersions + 1
+	newSchema := models.Schema{
+		ID:   schema.ID,
+		Name: schema.Name,
+		Type: schema.Type,
+	}
+
+	newSchemaVersion := models.SchemaVersion{
+		ID:            primitive.NewObjectID(),
+		VersionNumber: versionNumber,
+		Active:        false,
+		CreatedByUser: user.Username,
+		CreationDate:  time.Now(),
+		SchemaContent: schemaContent,
+		SchemaId:      newSchema.ID,
+	}
+
+	filter := bson.M{"schema_id": schema.ID, "version_number": newSchemaVersion.VersionNumber}
+	update := bson.M{
+		"$setOnInsert": bson.M{
+			"_id":             newSchemaVersion.ID,
+			"active":          newSchemaVersion.Active,
+			"created_by_user": newSchemaVersion.CreatedByUser,
+			"creation_date":   newSchemaVersion.CreationDate,
+			"schema_content":  newSchemaVersion.SchemaContent,
+		},
+	}
+
+	opts := options.Update().SetUpsert(true)
+	updateResults, err := schemaVersionCollection.UpdateOne(context.TODO(), filter, update, opts)
+	if err != nil {
+		serv.Errorf("CreateNewVersion error: " + err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+	if updateResults.MatchedCount == 0 {
+		message := "Version " + strconv.Itoa(newSchemaVersion.VersionNumber) + " has been created"
+		serv.Noticef(message)
+	} else {
+		serv.Warnf("Version already exists")
+		c.AbortWithStatusJSON(configuration.SHOWABLE_ERROR_STATUS_CODE, gin.H{"message": "Version already exists"})
+		return
+	}
+	c.IndentedJSON(200, newSchemaVersion)
+
+}
+
+func (sh SchemasHandler) RollBackVersion(c *gin.Context) {
+	var body models.RollBackVersion
+	ok := utils.Validate(c, &body, false, nil)
+	if !ok {
+		return
+	}
+
+	schemaName := strings.ToLower(body.SchemaName)
+	err := validateSchemaName(schemaName)
+	if err != nil {
+		serv.Warnf(err.Error())
+		c.AbortWithStatusJSON(configuration.SHOWABLE_ERROR_STATUS_CODE, gin.H{"message": err.Error()})
+		return
+	}
+	exist, _, err := IsSchemaExist(schemaName)
+	if err != nil {
+		serv.Errorf("RollBackVersion error: " + err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server Error"})
+		return
+	}
+	if !exist {
+		serv.Warnf("Schema does not exist")
+		c.AbortWithStatusJSON(configuration.SHOWABLE_ERROR_STATUS_CODE, gin.H{"message": "Schema does not exist"})
+		return
+	}
+
+	schemaVersion := body.VersionNumber
+	exist, _, err = isSchemaVersionExists(schemaVersion, schemaName)
+
+	if err != nil {
+		serv.Errorf("RollBackVersion error: " + err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+	if !exist {
+		serv.Warnf("Version does not exist")
+		c.AbortWithStatusJSON(configuration.SHOWABLE_ERROR_STATUS_CODE, gin.H{"message": "version does not exist"})
+		return
+	}
+
+	err = sh.updateActiveVersion(schemaName, body.VersionNumber)
```

Comment:
send here also the schema id instead of the name, this way you won't have to fetch it again inside thiss function